### ISSUE TITLE
Follow-up APIBindings: have more speaking names of CRDs in shadow namespace

### ISSUE THREAD

**Issue by sttts** [kcp-dev/kcp#843]
Follow-up of https://github.com/kcp-dev/kcp/issues/417.

Problem: the names of CRDs in the `system:bound-crds` workspace have cryptic names. For debugging it would be very helpful to have the resource and group name **plus** the uid there.

Also see https://github.com/kcp-dev/kcp/pull/702#discussion_r830045834 for details.

**Comment by sttts** [comment]
cc @ncdc

**Comment by ncdc** [comment]
What about having that data in labels or annotations? Or you think a more meaningful name is needed?

**Comment by ncdc** [comment]
Makes sense. Will resource+group+uid fit within the maximum name length limit?

**Comment by sttts** [comment]
Good question. Probably 63 chars for names. And that can be exhausted for a resource name alone I guess. But something like `resource.group-<8-alpha-numeric-chars>` is probably good enough. We can have a deterministic shortening of `resource.group` in case it's getting too long.
